storey/sources.py

```diff
@@ -1009,6 +1015,20 @@ def __init__(
 
             if filter_column is None:
                 raise TypeError("Filter column is required when passing start/end filters")
+        additional_filters = additional_filters or []
+
+        if not all(isinstance(item, tuple) for item in additional_filters):
+            raise ValueError(
+                f"ParquetSource supports additional_filters only as a list of tuples."
+                f" Current additional_filters: {additional_filters}"
+            )
+
+        for filters_tuple in additional_filters:
+            if filter_column and filter_column in filters_tuple:
+                raise ValueError(
+                    f"Cannot use the same column as both filter_column and in additional_filters."
+                    f" Column: {filter_column}."
+                )
```

Comment:
Why do we have to police this though? If the user really wants to, I don't see why we have to preempt them.